
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Genome Annotation &#8212; Apollo Portal  documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Contact" href="contact.html" />
    <link rel="prev" title="Data File Format" href="data_preparation.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="contact.html" title="Contact"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="data_preparation.html" title="Data File Format"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Apollo Portal  documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="genome-annotation">
<h1>Genome Annotation<a class="headerlink" href="#genome-annotation" title="Permalink to this headline">¶</a></h1>
<p>Jbrowse is a web-based genome visualisation and curation platform. The genome annotation can be generated using the annotation pipeline of interest. <a class="reference external" href="https://usegalaxy.org.au/">Galaxy</a> is a bioinformatics cloud platform containing a collection of bioinformatics for handling and processin genomic data.</p>
<div class="section" id="genome-annotation-using-galaxy">
<h2>Genome annotation using Galaxy<a class="headerlink" href="#genome-annotation-using-galaxy" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/MAKER_Tutorial_for_WGS_Assembly_and_Annotation_Winter_School_2018">Maker</a> is one of the popular genome annotation pipelines for annotating de novo assembly and is available on the main Galaxy (<a class="reference external" href="https://usegalaxy.org/">usegalaxy</a>).The <a class="reference external" href="https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">tutorial</a> of genome annotation with Maker contains a hands-on for genome annotation using Galaxy. Please see the Current Protocols (<a class="reference external" href="https://currentprotocols.onlinelibrary.wiley.com/doi/10.1002/0471250953.bi0411s48">CP</a>)  for more details.</p>
</div>
<div class="section" id="genome-annotation-data">
<h2>Genome annotation data<a class="headerlink" href="#genome-annotation-data" title="Permalink to this headline">¶</a></h2>
<p>Below is a list of genomic data files from the tutorial above.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">zenodo</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">api</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="mi">4385871</span><span class="n">d</span><span class="o">-</span><span class="mi">9632</span><span class="o">-</span><span class="mi">4</span><span class="n">fae</span><span class="o">-</span><span class="mi">9</span><span class="n">aaf</span><span class="o">-</span><span class="n">f8ed692163d1</span><span class="o">/</span><span class="n">augustus_training_1</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span>
<span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">zenodo</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">api</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="mi">4385871</span><span class="n">d</span><span class="o">-</span><span class="mi">9632</span><span class="o">-</span><span class="mi">4</span><span class="n">fae</span><span class="o">-</span><span class="mi">9</span><span class="n">aaf</span><span class="o">-</span><span class="n">f8ed692163d1</span><span class="o">/</span><span class="n">augustus_training_2</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span>
<span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">zenodo</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">api</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="mi">4385871</span><span class="n">d</span><span class="o">-</span><span class="mi">9632</span><span class="o">-</span><span class="mi">4</span><span class="n">fae</span><span class="o">-</span><span class="mi">9</span><span class="n">aaf</span><span class="o">-</span><span class="n">f8ed692163d1</span><span class="o">/</span><span class="n">S_pombe_chrIII</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">zenodo</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">api</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="mi">4385871</span><span class="n">d</span><span class="o">-</span><span class="mi">9632</span><span class="o">-</span><span class="mi">4</span><span class="n">fae</span><span class="o">-</span><span class="mi">9</span><span class="n">aaf</span><span class="o">-</span><span class="n">f8ed692163d1</span><span class="o">/</span><span class="n">S_pombe_genome</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">zenodo</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">api</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="mi">4385871</span><span class="n">d</span><span class="o">-</span><span class="mi">9632</span><span class="o">-</span><span class="mi">4</span><span class="n">fae</span><span class="o">-</span><span class="mi">9</span><span class="n">aaf</span><span class="o">-</span><span class="n">f8ed692163d1</span><span class="o">/</span><span class="n">S_pombe_trinity_assembly</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">zenodo</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">api</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="mi">4385871</span><span class="n">d</span><span class="o">-</span><span class="mi">9632</span><span class="o">-</span><span class="mi">4</span><span class="n">fae</span><span class="o">-</span><span class="mi">9</span><span class="n">aaf</span><span class="o">-</span><span class="n">f8ed692163d1</span><span class="o">/</span><span class="n">Swissprot_no_S_pombe</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>The Jbrowse GUI allows users to create/add a new organism by uploading the species file (e.g. FASTA) and the annotation file (e.g. gff3)
The Jbrowse requires both the raw and the index file to add new organism or tracks.</p>
<p>Please make sure the following files are prepared before the upload.</p>
<p>Organism file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">1</span><span class="p">)</span> <span class="n">fasta</span> <span class="n">file</span> <span class="n">ends</span> <span class="k">with</span> <span class="n">fa</span> <span class="ow">or</span> <span class="n">fasta</span> <span class="ow">and</span> <span class="n">index</span> <span class="n">file</span> <span class="n">ends</span> <span class="k">with</span> <span class="n">fai</span> <span class="p">(</span><span class="k">if</span> <span class="n">the</span> <span class="n">fasta</span> <span class="ow">is</span> <span class="n">added</span> <span class="n">using</span> <span class="n">command</span> <span class="n">line</span><span class="p">)</span><span class="o">.</span>
<span class="mi">2</span><span class="p">)</span> <span class="n">a</span> <span class="n">genome</span> <span class="n">file</span> <span class="ow">in</span> <span class="mi">2</span><span class="n">bit</span> <span class="nb">format</span> <span class="ow">is</span> <span class="n">required</span> <span class="k">for</span> <span class="n">the</span> <span class="n">upload</span> <span class="p">(</span><span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">species</span><span class="o">.</span><span class="mi">2</span><span class="n">bit</span><span class="p">)</span>
<span class="n">Samtools</span> <span class="n">can</span> <span class="n">be</span> <span class="n">used</span> <span class="n">to</span> <span class="n">create</span> <span class="n">the</span> <span class="n">index</span> <span class="n">file</span> <span class="k">with</span> <span class="n">the</span> <span class="n">following</span> <span class="n">command</span><span class="o">.</span>
<span class="n">The</span> <span class="mi">2</span><span class="n">bit</span> <span class="n">file</span> <span class="n">can</span> <span class="n">be</span> <span class="n">created</span> <span class="n">using</span> <span class="n">the</span> <span class="n">faToTwoBit_</span> <span class="n">tool</span><span class="o">.</span>
<span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span>
  <span class="n">samtools</span> <span class="n">faidx</span> <span class="n">species</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span>
  <span class="n">faToTwoBit</span> <span class="n">species</span><span class="o">.</span><span class="n">fa</span> <span class="n">species</span><span class="o">.</span><span class="mi">2</span><span class="n">bit</span>
</pre></div>
</div>
<p>Annotation file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Gff3</span> <span class="n">should</span> <span class="n">be</span> <span class="nb">sorted</span> <span class="ow">and</span> <span class="ow">is</span> <span class="n">required</span> <span class="n">to</span> <span class="n">be</span> <span class="n">compressed</span> <span class="n">using</span> <span class="n">the</span> <span class="n">bgzip</span> <span class="n">tool</span><span class="o">.</span>
<span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span>
  <span class="n">bgzip</span> <span class="n">species</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">gff3</span>
  <span class="n">tabix</span> <span class="o">-</span><span class="n">p</span> <span class="n">gff</span> <span class="n">species</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">gff3</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>Evidence Tracks (BAM) file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">The</span> <span class="n">evidence</span> <span class="n">track</span> <span class="n">file</span> <span class="n">requires</span> <span class="n">both</span> <span class="n">BAM</span> <span class="n">file</span> <span class="ow">and</span> <span class="n">BAM</span> <span class="n">index</span> <span class="n">file</span><span class="o">.</span>
<span class="n">Samtools</span> <span class="n">can</span> <span class="n">be</span> <span class="n">used</span> <span class="n">to</span> <span class="n">create</span> <span class="n">the</span> <span class="n">BAM</span> <span class="n">index</span> <span class="n">file</span> <span class="k">with</span> <span class="n">the</span> <span class="n">follwoing</span> <span class="n">command</span><span class="o">.</span>
<span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span>
  <span class="n">samtools</span> <span class="n">index</span> <span class="n">species</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>If the data size (&gt; 2G) can be made available on Jbrowse via command line. Please contact <a class="reference external" href="mailto:apollo-support&#37;&#52;&#48;genome&#46;edu&#46;au">us</a> after sign up on <a class="reference external" href="https://apollo-portal.genome.edu.au/">Apollo-portal</a> if you want to make your data available on Jbrowse via command line. Once you have granted the access to your Apollo VM, you can prepare your data in the following manner:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">1</span><span class="p">)</span> <span class="n">cd</span> <span class="n">into</span> <span class="n">your</span> <span class="n">apollo_data</span> <span class="n">folder</span>
<span class="mi">2</span><span class="p">)</span> <span class="n">create</span> <span class="n">a</span> <span class="n">folder</span> <span class="n">named</span> <span class="n">after</span> <span class="n">your</span> <span class="n">organism</span> <span class="p">(</span><span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">organims_A</span><span class="p">)</span>
<span class="mi">3</span><span class="p">)</span> <span class="n">cd</span> <span class="n">into</span> <span class="n">organism_A</span> <span class="n">folder</span>
<span class="mi">4</span><span class="p">)</span> <span class="n">index</span> <span class="n">the</span> <span class="n">organism_A</span><span class="o">.</span><span class="n">fasta</span> <span class="n">using</span> <span class="n">samtools</span> <span class="p">(</span><span class="n">see</span> <span class="n">the</span> <span class="n">instrunction</span> <span class="n">above</span><span class="p">)</span>
<span class="mi">5</span><span class="p">)</span> <span class="n">index</span> <span class="n">the</span> <span class="nb">sorted</span> <span class="n">annotation</span> <span class="n">file</span> <span class="p">(</span><span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">gff3</span> <span class="o">-</span> <span class="n">see</span> <span class="n">the</span> <span class="n">instruction</span> <span class="n">above</span><span class="p">)</span>
<span class="mi">6</span><span class="p">)</span> <span class="n">index</span> <span class="n">the</span> <span class="n">track</span> <span class="n">file</span> <span class="p">(</span><span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">BAM</span> <span class="o">-</span> <span class="n">see</span> <span class="n">the</span> <span class="n">instruction</span> <span class="n">above</span><span class="p">)</span>
<span class="mi">7</span><span class="p">)</span> <span class="n">run</span> <span class="n">prepare</span><span class="o">-</span><span class="n">refseqs</span><span class="o">.</span><span class="n">pl</span> <span class="o">--</span><span class="n">fasta</span> <span class="n">Organism_A</span><span class="o">.</span><span class="n">fa</span> <span class="n">to</span> <span class="n">prepare</span> <span class="n">the</span> <span class="n">genome</span> <span class="n">file</span> <span class="k">for</span> <span class="n">Jbrowse</span>
<span class="mi">8</span><span class="p">)</span> <span class="n">run</span> <span class="n">flatfile</span><span class="o">-</span><span class="n">to</span><span class="o">-</span><span class="n">json</span><span class="o">.</span><span class="n">pl</span> <span class="o">--</span><span class="n">tracklabel</span> <span class="n">Organism_A</span> <span class="o">--</span><span class="n">key</span> <span class="s2">&quot;Organism_A&quot;</span> <span class="o">--</span><span class="n">gff</span> <span class="n">Organism_A</span><span class="o">.</span><span class="n">gff3</span> <span class="o">--</span><span class="n">trackType</span> <span class="n">CanvasFeatures</span> <span class="o">--</span><span class="nb">type</span> <span class="n">CDS</span> <span class="o">--</span><span class="n">autocomplete</span> <span class="nb">all</span> <span class="n">to</span> <span class="n">prepare</span> <span class="n">annotation</span> <span class="n">file</span>
<span class="mi">9</span><span class="p">)</span> <span class="n">run</span> <span class="n">generate</span><span class="o">-</span><span class="n">names</span><span class="o">.</span><span class="n">pl</span> <span class="n">to</span> <span class="n">make</span> <span class="n">the</span> <span class="n">Organism_A</span> <span class="n">searchable</span> <span class="n">on</span> <span class="n">Jbrowse</span><span class="o">.</span>
<span class="n">Note</span><span class="p">:</span> <span class="n">the</span> <span class="n">perl</span> <span class="n">scripts</span> <span class="ow">is</span> <span class="n">part</span> <span class="n">of</span> <span class="ow">and</span> <span class="ow">is</span> <span class="n">available</span> <span class="n">on</span> <span class="n">apollo</span> <span class="n">platform</span><span class="o">.</span>
</pre></div>
</div>
<p>The folder layout</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">/</span><span class="n">Oragnism_A</span><span class="o">.</span><span class="n">fa</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">.</span><span class="n">fa</span><span class="o">.</span><span class="n">fai</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">gff3</span><span class="o">.</span><span class="n">gz</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">/</span><span class="n">Oragnism_A</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">gff3</span><span class="o">.</span><span class="n">gz</span><span class="o">.</span><span class="n">tbi</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">/</span><span class="n">Oragnism_A</span><span class="o">-</span><span class="nb">sorted</span><span class="o">.</span><span class="n">bam</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">-</span><span class="nb">sorted</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">bai</span>
<span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">apollo_data</span><span class="o">/</span><span class="n">Organism_A</span><span class="o">/</span><span class="n">tracks</span><span class="o">.</span><span class="n">conf</span>
</pre></div>
</div>
<p>The evidence and variant (if available) tracks can be made available by editing the tracks.conf file in the Organism_A folder in this example.</p>
<p>The tracks.conf file format.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="n">tracks</span><span class="o">.</span><span class="n">genes</span><span class="p">]</span>
<span class="n">urlTemplate</span><span class="o">=</span><span class="n">Organism_A</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">gff3</span><span class="o">.</span><span class="n">gz</span>
<span class="n">storeClass</span><span class="o">=</span><span class="n">JBrowse</span><span class="o">/</span><span class="n">Store</span><span class="o">/</span><span class="n">SeqFeature</span><span class="o">/</span><span class="n">GFF3Tabix</span>
<span class="nb">type</span><span class="o">=</span><span class="n">CanvasFeatures</span>

<span class="p">[</span><span class="n">tracks</span><span class="o">.</span><span class="n">alignments</span><span class="p">]</span>
<span class="n">urlTemplate</span><span class="o">=</span><span class="n">Organism_A</span><span class="o">-</span><span class="nb">sorted</span><span class="o">.</span><span class="n">bam</span>
<span class="n">storeClass</span><span class="o">=</span><span class="n">JBrowse</span><span class="o">/</span><span class="n">Store</span><span class="o">/</span><span class="n">SeqFeature</span><span class="o">/</span><span class="n">BAM</span>
<span class="nb">type</span><span class="o">=</span><span class="n">Alignments2</span>
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Genome Annotation</a><ul>
<li><a class="reference internal" href="#genome-annotation-using-galaxy">Genome annotation using Galaxy</a></li>
<li><a class="reference internal" href="#genome-annotation-data">Genome annotation data</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="data_preparation.html"
                        title="previous chapter">Data File Format</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="contact.html"
                        title="next chapter">Contact</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/genome_annotation.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="contact.html" title="Contact"
             >next</a> |</li>
        <li class="right" >
          <a href="data_preparation.html" title="Data File Format"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Apollo Portal  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2020, Apollo Portal.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.0.
    </div>
  </body>
</html>